Content.Client/Chameleon/ChameleonSystem.cs

```diff
@@ -62,12 +63,10 @@ private void OnShaderRender(EntityUid uid, SharedChameleonComponent component, B
         // unchanged.
         var parentXform = Transform(Transform(uid).ParentUid);
         var reference = args.Viewport.WorldToLocal(parentXform.WorldPosition);
-
+        var visibility = Getvisibility(uid, component);
         _shader.SetParameter("reference", reference);
-        _shader.SetParameter("stealthLevel", component.StealthLevel);
-        args.Sprite.Color = new Color(component.StealthLevel, component.StealthLevel, 1, 1);
-
-        Dirty(component);
```

Comment:
Theres no need to dirty components unless  you are actually changing networked variables (ie things that get sent in the component state).